src/modules/Dropdown/Dropdown.js

```diff
@@ -848,12 +861,12 @@ export default class Dropdown extends Component {
   // Setters
   // ----------------------------------------
 
-  setValue = (value) => {
+  setValue = (value, resetQuery = true) => {
```

Comment:
I think we should split this into two methods.  `setValue` only sets the value and `clearSearchQuery` only clears the query.  It is easier to test and reason about complex interactions when the methods are small and explicit.

There are also enough use cases to warrant seprating them I think:

### componentWillMount

This calls `setValue`, however, it calls it with `this.state.value`.  This is a redundant operation.  It seems the only reason we call it is for the other logic in `setValue`, clearing the search query and setting the selected index.

We don't need to clear the query on mount, but we do need to set the selected index.  I think this is duplicating work, but at the least, we don't need to clear the query on mount.

### componentWillReceiveProps

This calls `setValue` when a new value is received.  Though this is correct, I also don't think we want to clear the user's query when this happens.  If a new value comes down while a query is entered, I think it should only change the value, not the query.

### makeSelectedItemActive

This method currently takes a `resetQuery` argument and passes it down to `setValue`.  I think this method should only make the selected item active and not do anything else.  The call sites for this method should be responsible for also deciding to clear the query or not:

- `moveSelectionOnKeyDown` - should never clear the query
- `selectItemOnEnter` - should only clear query when !multiple
- `handleBlur` - should clear the query (it is already only called when !multiple)

### removeItemOnBackspace

Currently resets the query but is not necessary, there is no query at this point.  Also, if the user enters a query and moves the cursor to the begining of the query and then presses backspace, we wouldn't want to clear the query either:
```
[ LabelA x ] [ LabelB x ] | My query
                          ^ cursor, backspace should not clear the query
```

I know we don't support this yet, but we technically could/should in the future.

### handleItemClick

This currently resets the query when an menu item is made active, however, I have always found this to be a usability issue with `multiple` dropdowns.  When there are lots of items, it is often more helpful to keep the query while clicking items.  Otherwise, the user has to enter the same query over and over for every item.  Example, selecting all states that start with `ma`:

![http://g.recordit.co/OqqnZDHR5u.gif](http://g.recordit.co/OqqnZDHR5u.gif)

If we separate `setValue` and `clearSearchQuery`, then we can call only `setValue` in the multiple condition in this method.  The other branch (not multiple) we can safely clear the query becuase we know the user is done making selections.

### handleLabelRemove

This will currently clear the query also but probably never should have done so.  Here I'm again selecting states with `ma`.  I "accidentally" select `Alabama` and continue making selections.  I then remove `Alabama` while I have a query entered and my query gets removed :/

![http://g.recordit.co/IqdAPXNVrQ.gif](http://g.recordit.co/IqdAPXNVrQ.gif)